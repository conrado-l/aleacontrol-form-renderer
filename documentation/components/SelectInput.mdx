---
name: SelectInput
---

import { Playground, PropsTable } from 'docz'
import SelectInput from '../../src/components/SelectInput.js'

# SelectInput

<PropsTable of={SelectInput} />

## Description
A radio input with options
## Demo and usage

<Playground>
    <SelectInput value='male' name='gender' options={[{value: 'male', description: 'Male'}, {value: 'female', description: 'Female'}]} col={4} update={(e) => e}/>
</Playground>
