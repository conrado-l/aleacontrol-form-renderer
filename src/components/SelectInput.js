import React, { Component } from 'react'
import PropTypes from 'prop-types'

class SelectInput extends Component {
  render () {
    // Generate the select options
    const options = this.props.options.map((input) => {
      return <option value={input.value} key={input.value}> {input.description}</option>
    })

    return (
      <select className={`form-control col-xs-12 col-${this.props.col}`} onChange={(e) => this.props.update(this.props.name, e.target.value)}>
        {options}
      </select>
    )
  }
}

SelectInput.propTypes = {
  options: PropTypes.array,
  name: PropTypes.string,
  col: PropTypes.number,
  update: PropTypes.func
}
export default SelectInput
