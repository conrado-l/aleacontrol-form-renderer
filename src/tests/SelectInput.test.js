import React from 'react'
import { shallow, mount } from 'enzyme'
import SelectInput from '../components/SelectInput'
import renderer from 'react-test-renderer'

describe('SelectInput', () => {
  const mockupProps = {
    name: 'continent',
    value: 'europe',
    options: [
      { value: 'asia', description: 'Asia' },
      { value: 'europe', description: 'Europe' }]
  }

  it('should render correctly select component', () => {
    const SelectInputComponent = renderer
      .create(<SelectInput name={mockupProps.name} value={mockupProps.value} options={mockupProps.options} />)
      .toJSON()
    expect(SelectInputComponent).toMatchSnapshot()
  })

  it('should render the select options', () => {
    const component = shallow(<SelectInput value={mockupProps.value} options={mockupProps.options} />)

    expect(
      component.containsMatchingElement(
        <option value={mockupProps.options[0].value}>{mockupProps.options[0].description}</option>

      )
    ).toBeTruthy()

    expect(
      component.containsMatchingElement(
        <option value={mockupProps.options[1].value}>{mockupProps.options[1].description}</option>
      )
    ).toBeTruthy()
  })

  it('should call the update callback function with the proper values on change', () => {
    const changFn = jest.fn()
    const component = mount(<SelectInput update={changFn} name={mockupProps.name} value={mockupProps.value}
      options={mockupProps.options} />)
    component.find('select').simulate('change', {
      target: {
        value: component.props().options[0].value
      }
    })

    expect(changFn).toHaveBeenCalledWith(mockupProps.name, component.props().options[0].value)
  })
})
