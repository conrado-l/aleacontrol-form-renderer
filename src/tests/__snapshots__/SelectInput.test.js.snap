// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectInput should render correctly select component 1`] = `
<div
  className="row"
>
  <div
    className="col-12"
  >
    <select
      className="form-control"
      defaultValue="europe"
      onChange={[Function]}
    >
      <option
        value="asia"
      >
         
        Asia
      </option>
      <option
        value="europe"
      >
         
        Europe
      </option>
    </select>
  </div>
</div>
`;
